plugins/summoner.js

```diff
@@ -46,6 +46,9 @@ module.exports = function summoner(forum, config) {
         debug('summoner received a mention notification!');
         return notification.getUser()
             .then((user) => {
+                if (user.username && !user.name) {
```

Comment:
Agreed. For now though, I think this achieves what I intended. I can always go back and implement the more extensible version later 🙂